File: frontend/src/types/appointment.ts
export type AppointmentStatus =
  | "PENDING"
  | "CONFIRMED"
  | "IN_PROGRESS"
  | "COMPLETED"
  | "CANCELLED"
  | "NO_SHOW";

export interface AppointmentCustomerSummary {
  id: string;
  full_name: string;
  phone: string;
}

export interface AppointmentEmployeeSummary {
  id: string;
  full_name: string;
  role_title: string | null;
}

export interface AppointmentServiceSummary {
  id: string;
  name: string;
  price: string;
  duration_minutes: number;
}

export interface Appointment {
  id: string;
  customer: AppointmentCustomerSummary;
  employee: AppointmentEmployeeSummary;
  service: AppointmentServiceSummary;
  starts_at: string;
  ends_at: string;
  duration_minutes: number;
  price: string;
  payment_method: string | null;
  status: AppointmentStatus;
  notes: string | null;
  cancellation_reason: string | null;
  created_at: string;
}

export interface AppointmentPayload {
  customer_id: string;
  employee_id: string;
  service_id: string;
  starts_at: string;
  notes?: string | null;
}

export interface ReschedulePayload {
  starts_at: string;
  employee_id?: string;
}

export interface ProductSaleItem {
  product_id: string;
  quantity: number;
}

export interface CompletePayload {
  price?: string;
  payment_method_code: string;
  products_sold?: ProductSaleItem[];
}

File: backend/app/schemas/appointment.py
import uuid
from datetime import datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, Field

from app.models.appointment import AppointmentStatus
from app.schemas.customer import CustomerSummary
from app.schemas.employee import EmployeeSummary
from app.schemas.service import ServiceSummary


class AppointmentCreate(BaseModel):
    customer_id: uuid.UUID
    employee_id: uuid.UUID
    service_id: uuid.UUID
    starts_at: datetime
    notes: str | None = None


class AppointmentReschedule(BaseModel):
    starts_at: datetime
    employee_id: uuid.UUID | None = None


class AppointmentCancel(BaseModel):
    reason: str = Field(min_length=3, max_length=500)


class ProductSaleItem(BaseModel):
    product_id: uuid.UUID
    quantity: int = Field(gt=0)


class AppointmentComplete(BaseModel):
    price: Decimal | None = Field(default=None, ge=0)
    payment_method_code: str
    products_sold: list[ProductSaleItem] = Field(default_factory=list)


class AppointmentRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    customer: CustomerSummary
    employee: EmployeeSummary
    service: ServiceSummary
    starts_at: datetime
    ends_at: datetime
    duration_minutes: int
    price: Decimal
    payment_method: str | None
    status: AppointmentStatus
    notes: str | None
    cancellation_reason: str | None
    created_at: datetime


class AppointmentStatusHistoryRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    from_status: AppointmentStatus
    to_status: AppointmentStatus
    reason: str | None
    changed_by_user_id: uuid.UUID | None
    created_at: datetime
